cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/artifact/RemoteArtifactRepositoryReader.java

```diff
@@ -87,6 +92,30 @@ public ArtifactDetail getArtifact(Id.Artifact artifactId) throws Exception {
     return artifactDetail;
   }
 
+  @Override
+  public List<ArtifactDetail> getArtifactDetails(ArtifactRange range, int limit, ArtifactSortOrder order)
+    throws Exception {
+    String url = String.format("namespaces/%s/artifacts/%s/versions?lower=%s&upper=%s&limit=%s&order=%s",
+                               range.getNamespace(),
+                               range.getName(),
+                               range.getLower().toString(),
+                               range.getUpper().toString(),
+                               String.valueOf(limit),
+                               String.valueOf(order));
+    HttpRequest.Builder requestBuilder = remoteClient.requestBuilder(HttpMethod.GET, url);
+    HttpResponse httpResponse = execute(requestBuilder.build());
+    List<ArtifactDetail> details = GSON.fromJson(httpResponse.getResponseBodyAsString(), ARTIFACT_DETAIL_LIST_TYPE);
+    List<ArtifactDetail> detailList = new ArrayList<>();
+    for (ArtifactDetail detail : details) {
+      Location artifactLocation =
+        Locations.getLocationFromAbsolutePath(locationFactory, detail.getDescriptor().getLocationURI().getPath());
```

Comment:
is there a reason this utility method is being used over locationFactory.create()? Are they different?

Reply:
Fixed. We can use LocationFactory.create() directly. The util cleans up and validate the path. But we are guaranteed the location URI is valid, so can just use create.